sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -69,6 +69,33 @@ object CosmosAccountConfig {
 
 case class CosmosContainerConfig(database: String, container: String)
 
+case class CosmosWriteConfig(upsertEnabled: Boolean, maxRetryCount: Int)
+
+object CosmosWriteConfig {
+  val upsertEnabled = CosmosConfigEntry[Boolean](key = "spark.cosmos.write.upsertEnabled",
+    mandatory = false,
+    defaultValue = Option.apply(true), // TODO: what the default value should be?
+    parseFromStringFunction = overwriteEnabled => overwriteEnabled.toBoolean,
+    helpMessage = "Cosmos DB Write Upsert Enabled")
+
+  val maxRetryCount = CosmosConfigEntry[Int](key = "spark.cosmos.write.maxRetryCount",
+    mandatory = false,
+    defaultValue = Option.apply(3),
+    parseFromStringFunction = maxRetryAttempt => maxRetryAttempt.toInt,
+    helpMessage = "Cosmos DB Write Max Retry Attempts on failure")
+
+  def parseWriteConfig(cfg: Map[String, String]): CosmosWriteConfig = {
+    val upsertEnabledOpt = CosmosConfigEntry.parse(cfg, upsertEnabled)
+    val maxRetryCountOpt = CosmosConfigEntry.parse(cfg, maxRetryCount)
+
+    // parsing above already validated this
+    assert(upsertEnabledOpt.isDefined)
+    assert(maxRetryCountOpt.isDefined)
```

Comment:
validate the retry count is positive integer?

Reply:
good suggestion. we should. I will add. Thanks Matias.

Reply:
addressed. thanks